doc: CPLYTM-974 Add doc for testing farm tests (#282)
complytime/complyctl: docs/TESTING_FARM.md

# Tests using Testing Farm 

> [Testing frame](https://packit.dev/docs/configuration/upstream/tests) is Packit's testing system.

Reviewer: ```suggestion
[Testing Farm](https://packit.dev/docs/configuration/upstream/tests) is Packit's testing system.
```

Reviewer: It was intentional to use "Testing frame" or was the idea to use "Testing Farm"? Just to double-check.

Test execution is managed by tmt tool. 

> The entry of testing frame tests located at [.packit.yaml](../.packit.yaml), check the job named `tests`.

Reviewer: ```suggestion
The entry of the testing farm tests is located at [.packit.yaml](../.packit.yaml), in the the job named `tests`.
```

The `tests` job requires `copr_build` job to be built before running tests, 

> and then it is automatically installed built packages into the testing environment.

Reviewer: ```suggestion
so the built packages are automatically installed in the testing environment.
```

> If you want to add new tests or extend existing tests, follow [Testing frame doc](https://packit.dev/docs/configuration/upstream/tests)

Reviewer: ```suggestion
The [Testing Farm documentation](https://packit.dev/docs/configuration/upstream/tests) gives information on how to include or modify tests.
```

## Test RPM provided OSCAL content 

> Currently, we have a [simple test](../plans/test-RPM-provide-content.fmf) to make sure our RPM package 

Reviewer: ```suggestion
Currently, there is a [simple test](../plans/test-RPM-provide-content.fmf) to make sure the RPM package 
```

> provided OSCAL content can be used out of the box. 

Reviewer: ```suggestion
provides OSCAL content that can be used out of the box. 
```
